.github/workflows/deploy.yml

```diff
@@ -0,0 +1,32 @@
+name: Deploy to pages.dev
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    env:
+      CLOUDFLARE_API_TOKEN: "${{secrets.CLOUDFLARE_API_TOKEN}}"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+
+      - name: Deploy to Pages.dev using Wrangler
+        run: |
+          npx -y wrangler pages deploy \
+            --project-name gps-route-tracker \
+            --branch ${{github.ref_name == 'master' && 'main' && github.ref_name}} \
+            .
```

Comment:
My default branch name is master, but Cloudflare presumes it's `main`, so in case we are running the job for master, we need to rename it to `main` for Cloudflare's purposes.